### PR TITLE
Tdd walkthrough

### PR DESCRIPTION
**Pull request by tjyoung99**
Pull request for Test Driven Development Walk-through.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,65 @@
...
+import java.util.Set;
+import com.google.sps.Events;
+import com.google.sps.TimeRange;
+
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Remove this comment

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,65 @@
...
+    Collection<TimeRange> result = new ArrayList<TimeRange>();
+      if (request.getDuration()> 24* 60) {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I would use a descriptive constant for 24 * 60 and also add space after the 24
Also add space after getDuration()

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,65 @@
...
+        return Arrays.asList();
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: You can return result here

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -14,10 +14,65 @@
...
+      }
+    if(events.isEmpty() || request.getAttendees().isEmpty() || request.getAttendees().equals(getAllAttendees(events))){
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    List<TimeRange> listOfRanges = new ArrayList<TimeRange>();
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Initialize this to the same capacity as events to avoid ArrayList resizing as you add elements

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -14,10 +14,65 @@
...
+    }
+    for (int i = 0; i < listOfRanges.size(); i++) {
+      if (i == listOfRanges.size()-1) {
+        availableTime = TimeRange.fromStartEnd(listOfRanges.get(i).end(), TimeRange.END_OF_DAY, true);
+        if (availableTime.duration() >= request.getDuration()) {
+          result.add(availableTime);
+          break;
+        }   
+      }
+      else {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit. move else to the above line

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 6]
```diff
@@ -14,10 +14,65 @@
...
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> result = new ArrayList<TimeRange>();
+      if (request.getDuration()> 24* 60) {
+        return Arrays.asList();
+      }
+    if(events.isEmpty() || request.getAttendees().isEmpty() || request.getAttendees().equals(getAllAttendees(events))){
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    List<TimeRange> listOfRanges = new ArrayList<TimeRange>();
+    for(Event event : events){
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit add space after for

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 7]
```diff
@@ -14,10 +14,65 @@
...
+      listOfRanges.add(event.getWhen());
+    }
+    Collections.sort(listOfRanges,TimeRange.ORDER_BY_START);
+    TimeRange availableTime;
+    TimeRange startOfDay = TimeRange.fromStartEnd(TimeRange.START_OF_DAY,listOfRanges.get(0).start(), false);
+    if (startOfDay.duration() >= request.getDuration()) {
+      result.add(startOfDay);
+    }
+    for (int i = 0; i < listOfRanges.size(); i++) {
+      if (i == listOfRanges.size()-1) {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit. add space after .size()
i == listOfRanges.size() - 1

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 8]
```diff
@@ -14,10 +14,65 @@
...
+    if(events.isEmpty() || request.getAttendees().isEmpty() || request.getAttendees().equals(getAllAttendees(events))){
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    List<TimeRange> listOfRanges = new ArrayList<TimeRange>();
+    for(Event event : events){
+      listOfRanges.add(event.getWhen());
+    }
+    Collections.sort(listOfRanges,TimeRange.ORDER_BY_START);
+    TimeRange availableTime;
+    TimeRange startOfDay = TimeRange.fromStartEnd(TimeRange.START_OF_DAY,listOfRanges.get(0).start(), false);
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit. add spaces between the arguments

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 9]
```diff
@@ -14,10 +14,65 @@
...
+        availableTime = TimeRange.fromStartEnd(listOfRanges.get(i).end(), listOfRanges.get(i+1).start(), false);
+        if(availableTime.duration() >= request.getDuration()) {
+          result.add(availableTime);
+        }   
+      }
+    }
+    System.out.println(request.getAttendees().equals(getAllAttendees(events)))
+    System.out.println(result);
+    return result;
+  }
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:90]: nit. add new line here

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 10]
```diff
@@ -14,10 +14,65 @@
...
+  public Set<String> getAllAttendees(Collection<Event> events) {
+    Set<String> result = new HashSet<String>();
+    for(Event event: events){
+      Set<String> attendees = event.getAttendees();
+      for(String attendee: attendees) {
+        result.add(attendee);
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: you can just do result.addAll(event.getAttendees())

**Review by suparnabarua — CHANGES_REQUESTED** [review]
First round of comments

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 11]
```diff
@@ -14,10 +14,112 @@
...
+
+    List<TimeRange> RangesByStart = new ArrayList<TimeRange>(events.size());
+    for (Event event : events){
+      RangesByStart.add(event.getWhen());
+    }
+    List<TimeRange> RangesByEnd = new ArrayList<TimeRange>(RangesByStart);
+    Collections.sort(RangesByStart, TimeRange.ORDER_BY_START);
+    Collections.sort(RangesByEnd, TimeRange.ORDER_BY_END);
+    
+    int length = RangesByStart.size();
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit. rename this to rangeLength or rangeSize

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 12]
```diff
@@ -14,10 +14,112 @@
...
+  
+  public Set<String> getAllAttendees(Collection<Event> events) {
+    Set<String> allAttendees = new HashSet<String>();
+    for (Event event: events){
+      allAttendees.addAll(event.getAttendees());
+    }
+    return allAttendees;
+  }
+
+  public boolean collectionContainsSet(Collection<String> stringCollection, Set<String> stringSet) {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: This could be replaced with the Java Set.containsAll method

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 13]
```diff
@@ -14,10 +14,112 @@
...
+   
+    TimeRange requestableTime = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, RangesByStart.get(0).start(), false);
+
+    if (requestableTime.duration() >= request.getDuration()) {
+      collectionOfRanges.add(requestableTime);
+    }
+
+     requestableTime = 
+        TimeRange.fromStartEnd(RangesByEnd.get(length - 1).end(), TimeRange.END_OF_DAY, true);
+    if (requestableTime.duration() >= request.getDuration()) {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: You're repeating code here. This could be moved into a helper function that takes requestableTime, request.getDuration and collectionOfRanges.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 14]
```diff
@@ -14,10 +14,112 @@
...
+      collectionOfRanges.add(requestableTime);
+    }
+
+    TimeRange currentRange;
+    TimeRange nextRange;
+    List<TimeRange> overlappingRanges;
+    if (length > 1) {
+      for (int i = 0; i < length; i++) {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: This block should be a helper function as it's gotten really big.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 15]
```diff
@@ -14,10 +14,112 @@
...
+          nextRange = RangesByStart.get(i + 1);
+          TimeRange latestRange = overlappingRanges.get(0);
+          requestableTime = TimeRange.fromStartEnd(
+              latestRange.end(), 
+              nextRange.start(), 
+              false);
+    
+          if (requestableTime.duration() >= request.getDuration()) {
+            collectionOfRanges.add(requestableTime);
+            continue;
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: The continue here is redundant

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 16]
```diff
@@ -14,10 +14,112 @@
...
+          i = j;
+          if (i + 1 >= length) break;
+          nextRange = RangesByStart.get(i + 1);
+          TimeRange latestRange = overlappingRanges.get(0);
+          requestableTime = TimeRange.fromStartEnd(
+              latestRange.end(), 
+              nextRange.start(), 
+              false);
+    
+          if (requestableTime.duration() >= request.getDuration()) {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: You're doing the same check in the else block as well. Instead of repeating it in both the if and else blocks, have it be outside the if/else and within the for loop

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 17]
```diff
@@ -14,10 +14,112 @@
...
+    if (length > 1) {
+      for (int i = 0; i < length; i++) {
+        currentRange = RangesByStart.get(i);
+        if (i + 1 >= length) break;
+        nextRange = RangesByStart.get(i + 1);
+        if (currentRange.overlaps(nextRange)){
+          overlappingRanges = new ArrayList<TimeRange>(length);
+          int j = 0;
+          for (TimeRange range : RangesByStart) {
+            j++;
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Isn't j always going to be the siz eof RangesByStart or length?

**Review by suparnabarua — CHANGES_REQUESTED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 18]
```diff
@@ -14,10 +14,110 @@
...
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import com.google.sps.Events;
+import com.google.sps.TimeRange;
+
+
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit: Extra lines

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 19]
```diff
@@ -14,10 +14,110 @@
...
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> collectionOfRanges = new ArrayList<TimeRange>();
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return collectionOfRanges;
+    }
+
+    if (events.isEmpty() || request.getAttendees().isEmpty() || 
+        !(getAllAttendees(events).containsAll(request.getAttendees()))) {
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: This isn't necessarily the case, right?  For instance, maybe attendee A doesn't have any events (i.e. free the whole day), but attendee B does have events (i.e. not free the whole day).  In this scenario `getAllAttendees(events)` would not contain all request attendees, because it doesn't contain attendee A.  However, attendee B still isn't free the whole day.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 20]
```diff
@@ -14,10 +14,110 @@
...
+    return collectionOfRanges;
+  }
+  
+  public Set<String> getAllAttendees(Collection<Event> events) {
+    Set<String> allAttendees = new HashSet<String>();
+    for (Event event: events){
+      allAttendees.addAll(event.getAttendees());
+    }
+    return allAttendees;
+  }
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: What's the purpose of this function? Why does it matter who "all" the attendees are?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 21]
```diff
@@ -14,10 +14,110 @@
...
+    if (events.isEmpty() || request.getAttendees().isEmpty() || 
+        !(getAllAttendees(events).containsAll(request.getAttendees()))) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    List<TimeRange> rangesByStart = new ArrayList<TimeRange>(events.size());
+    for (Event event : events){
+      rangesByStart.add(event.getWhen());
+    }
+    List<TimeRange> rangesByEnd = new ArrayList<TimeRange>(rangesByStart);
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: How do you know that all of these time ranges are relevant to the request?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 22]
```diff
@@ -14,10 +14,110 @@
...
+    for (TimeRange range : listOfRanges) {
+      if (currentRange.overlaps(range)) {
+        overlappingRanges.add(range);
+      }
+    }
+    Collections.sort(overlappingRanges, TimeRange.ORDER_BY_END);
+    return overlappingRanges;
+  }
+  
+  public void addAllOtherTimeRanges(List<TimeRange> rangesByStart, TimeRange requestableTime, MeetingRequest request, List<TimeRange> collectionOfRanges, int rangesLength){
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit: Is this line under 100 characters?

**Review by kellyktan — CHANGES_REQUESTED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 22 above)
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Break up long lines

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 23]
```diff
@@ -14,10 +14,110 @@
...
+    List<TimeRange> overlappingRanges;
+    for (int i = 0; i < rangesLength - 1; i++) {
+        currentRange = rangesByStart.get(i);
+        nextRange = rangesByStart.get(i + 1);
+        if (currentRange.overlaps(nextRange)){
+          overlappingRanges = getOverlappingRanges(currentRange, rangesByStart);
+          i = rangesByStart.indexOf(overlappingRanges.get(0));
+          if (i + 1 >= rangesLength) break;
+          nextRange = rangesByStart.get(i + 1);
+          TimeRange latestRange = overlappingRanges.get(0);
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: So latestRange here is equivalent to currentRange=rangesByStart.get(i) right?  In that case you don't need the latestRange variable and can also move the requestableTime = TimeRange...... operation to the common part of the for loop and get rid of the else block. Does that make sense?

**Review by suparnabarua — CHANGES_REQUESTED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 19 above)
Comment (tjyoung99) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: getAllAttendees() will only contain B since A is free all day and if B is requested then request.getattendees() will also only contain B. Therefore, with only B being requested, the condition will be true but it will be reversed by the not operator. if a and b are selected it will do the same and lastly if only a is selected it will return false which is reversed to true and it will return whole day.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 20 above)
Comment (tjyoung99) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: it really returns all attendees involved in events I will change the name and add a comment to make this more clear.
We need to know who are in each event because we will check this in a condition to make sure that we only account for people that are requested and not just everyone in every event.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 21 above)
Comment (tjyoung99) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: so this wasn't considered as this wasn't an edge case tested by the provided tests. so I added it as a test and it definitely failed. I added a condition to check to see if attendees in the event were requested before adding to the list and now the new test passes.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 24]
```diff
@@ -14,10 +14,109 @@
...
+   
+    TimeRange requestableTime = 
+        TimeRange.fromStartEnd(TimeRange.START_OF_DAY, rangesByStart.get(0).start(), false);
+    addTimeRange(requestableTime, request, collectionOfRanges);
+
+     requestableTime = 
+        TimeRange.fromStartEnd(rangesByEnd.get(rangesLength - 1).end(), TimeRange.END_OF_DAY, true);
+    addTimeRange(requestableTime, request, collectionOfRanges);
+
+
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:58]: nit. delete extra line

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 19 above)
Comment (kellyktan) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: If A & B are requested, `getAllAttendees(events).containsAll([A, B])` would evaluate to `false` (since A does not have events), which would be reversed to `true` by the `!`.  Then this would return `WHOLE_DAY` even though B is not available the whole day.

Comment (tjyoung99) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Ahh I understand now. I wrote another test to demonstrate this edge case, and I altered a method so that it would check whether requested have events instead of whether the event attendees are in the requested list.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 25]
```diff
@@ -67,12 +67,16 @@
...
-    for (Event event: events){
+    Set<String> requestedAttendees = new HashSet<String>(request.getAttendees());
+    for (Event event : events){
       allAttendees.addAll(event.getAttendees());
     }
-    return allAttendees;
+    for (String attendee : requestedAttendees) {
+      if (allAttendees.contains(attendee)) return false;
+    }
+    return true;
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: You can use the `Collections.disjoint()` method for comparing collections (lists, sets, etc.)

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 26]
```diff
@@ -14,10 +14,112 @@
...
+      if (currentRange.overlaps(range)) {
+        overlappingRanges.add(range);
+      }
+    }
+    Collections.sort(overlappingRanges, TimeRange.ORDER_BY_END);
+    return overlappingRanges;
+  }
+  
+  public void addAllOtherTimeRanges(List<TimeRange> rangesByStart, 
+                                    TimeRange requestableTime,
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit dont make every argument a different line

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 27]
```diff
@@ -14,10 +14,112 @@
...
+    }
+    Collections.sort(overlappingRanges, TimeRange.ORDER_BY_END);
+    return overlappingRanges;
+  }
+  
+  public void addAllOtherTimeRanges(List<TimeRange> rangesByStart, 
+                                    TimeRange requestableTime, 
+                                    MeetingRequest request, 
+                                    List<TimeRange> collectionOfRanges, 
+                                    int rangesLength) {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Do you need to pass rangesLength since it should be rangesByStart.size()?

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 28]
```diff
@@ -270,5 +271,44 @@ public void notEnoughRoom() {
...
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+                      TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+                      TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void somePeopleAre() {
```
Comment (suparnabarua) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: rename this test name to describe what condition you're testing

**Review by suparnabarua — CHANGES_REQUESTED** [review]

**Review by suparnabarua — APPROVED** [review]
LGTM pending open comments

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 29]
```diff
@@ -73,10 +69,7 @@ public boolean ifEventAttendeesNotRequested(Collection<Event> events, MeetingReq
     for (Event event : events){
       allAttendees.addAll(event.getAttendees());
     }
-    for (String attendee : requestedAttendees) {
-      if (allAttendees.contains(attendee)) return false;
-    }
-    return true;
+    return Collections.disjoint(allAttendees,requestedAttendees);
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit: space after comma

You can probably use `request.getAttendees()` here directly?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 30]
```diff
@@ -65,11 +65,10 @@
   
   public boolean ifEventAttendeesNotRequested(Collection<Event> events, MeetingRequest request) {
     Set<String> allAttendees = new HashSet<String>();
-    Set<String> requestedAttendees = new HashSet<String>(request.getAttendees());
     for (Event event : events){
       allAttendees.addAll(event.getAttendees());
     }
-    return Collections.disjoint(allAttendees,requestedAttendees);
+    return Collections.disjoint(allAttendees,request.getAttendees());
```
Comment (kellyktan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:71]: nit: space after comma

**Review by kellyktan — APPROVED** [review]
